process/block/preprocess/transactionsV2.go

```diff
@@ -0,0 +1,416 @@
+			"sc txs", mapScsForShard[miniBlock.ReceiverShardID])
+	}
+
+	log.Debug("createAndProcessMiniBlocksFromMeV2 has been finished",
+		"total txs", len(sortedTxs),
+		"num txs added", numTxsAdded,
+		"num txs bad", numTxsBad,
+		"num txs failed", numTxsFailed,
+		"num txs skipped", numTxsSkipped,
+		"num txs with initial balance consumed", numTxsWithInitialBalanceConsumed,
+		"num cross shard sc calls or special txs", numCrossShardScCallsOrSpecialTxs,
+		"used time for computeGasConsumed", totalTimeUsedForComputeGasConsumed,
+		"used time for processAndRemoveBadTransaction", totalTimeUsedForProcesss)
+
+	return miniBlocks, nil
+}
+
+func (txs *transactions) getMiniBlockSliceFromMapV2(
+	mapMiniBlocks map[uint32]*block.MiniBlock,
+	mapSCTxs map[string]struct{},
+) block.MiniBlockSlice {
+	miniBlocks := make(block.MiniBlockSlice, 0)
+
+	for shardID := uint32(0); shardID < txs.shardCoordinator.NumberOfShards(); shardID++ {
+		if miniBlock, ok := mapMiniBlocks[shardID]; ok {
+			if len(miniBlock.TxHashes) > 0 {
+				miniBlocks = append(miniBlocks, splitMiniBlockIfNeeded(miniBlock, mapSCTxs)...)
+			}
+		}
+	}
+
+	if miniBlock, ok := mapMiniBlocks[core.MetachainShardId]; ok {
+		if len(miniBlock.TxHashes) > 0 {
+			miniBlocks = append(miniBlocks, splitMiniBlockIfNeeded(miniBlock, mapSCTxs)...)
+		}
+	}
+
+	return miniBlocks
+}
+
+func splitMiniBlockIfNeeded(miniBlock *block.MiniBlock, mapSCTxs map[string]struct{}) block.MiniBlockSlice {
+	splitMiniBlocks := make(block.MiniBlockSlice, 0)
+	notScTxHashes := make([][]byte, 0)
+	scTxHashes := make([][]byte, 0)
+
+	for _, txHash := range miniBlock.TxHashes {
+		_, isSCTx := mapSCTxs[string(txHash)]
+		if !isSCTx {
+			notScTxHashes = append(notScTxHashes, txHash)
+			continue
+		}
+
+		scTxHashes = append(scTxHashes, txHash)
+	}
+
+	if len(notScTxHashes) > 0 {
+		notScMiniBlock := &block.MiniBlock{
+			TxHashes:        notScTxHashes,
+			SenderShardID:   miniBlock.SenderShardID,
+			ReceiverShardID: miniBlock.ReceiverShardID,
+			Type:            miniBlock.Type,
+			Reserved:        miniBlock.Reserved,
+		}
+
+		splitMiniBlocks = append(splitMiniBlocks, notScMiniBlock)
+	}
+
+	if len(scTxHashes) > 0 {
+		scMiniBlock := &block.MiniBlock{
+			TxHashes:        scTxHashes,
+			SenderShardID:   miniBlock.SenderShardID,
+			ReceiverShardID: miniBlock.ReceiverShardID,
+			Type:            miniBlock.Type,
+			Reserved:        miniBlock.Reserved,
+		}
+
+		splitMiniBlocks = append(splitMiniBlocks, scMiniBlock)
+	}
+
+	return splitMiniBlocks
+}
+
+//TODO: Finish implementation of this method which should create all the scheduled mini blocks if needed
+func (txs *transactions) createScheduledMiniBlocks(
+	haveTime func() bool,
+	isShardStuck func(uint32) bool,
+	isMaxBlockSizeReached func(int, int) bool,
+	sortedTxs []*txcache.WrappedTransaction,
+	mapSCTxs map[string]struct{},
+) block.MiniBlockSlice {
+	log.Debug("createScheduledMiniBlocks has been started")
+
+	mapMiniBlocks := make(map[uint32]*block.MiniBlock)
```

Comment:
take out into a function lines 365 and 371

Reply:
Done